### DISCUSSION TITLE
Enabling server for Metal inference (apple silicon only)

### DISCUSSION THREAD

**Discussion by x4080** [ggml-org/llama.cpp#1722] [Q&A] [answered]
Hi all,

with this Metal inference (apple silicon only) feature from https://github.com/ggerganov/llama.cpp/pull/1642/
how to using it with server ?

I already try it and it seems not plug & play for server, how to modify the server to enable this feature ?

Thanks

@FSSRepo Sorry to bother you again brother :)

**Comment by FSSRepo** [comment, marked as answer]
Hello, you need config the project with metal

```bash
cd llama.cpp/build
```

Config cmake to build with metal option:

```bash
cmake .. -DLLAMA_METAL=ON
cmake --build . --config Release
```

Run the server with gpu acceleration:

```bash
./server -m modelfile -ngl 1
```

change the ngl to offload a number of layers to the gpu

**Comment by x4080** [comment]
@FSSRepo thanks, I never thought it will be that simple, I'll try it out

Edit : Wow it just works
